crates/simplexpr/src/eval.rs

```diff
@@ -232,24 +232,31 @@ impl SimplExpr {
 
                 let is_safe = *safe == AccessType::Safe;
 
-                match val.as_json_value()? {
-                    serde_json::Value::Array(val) => {
-                        let index = index.as_i32()?;
-                        let indexed_value = val.get(index as usize).unwrap_or(&serde_json::Value::Null);
-                        Ok(DynVal::from(indexed_value).at(*span))
-                    }
-                    serde_json::Value::Object(val) => {
-                        let indexed_value = val
-                            .get(&index.as_string()?)
-                            .or_else(|| val.get(&index.as_i32().ok()?.to_string()))
-                            .unwrap_or(&serde_json::Value::Null);
-                        Ok(DynVal::from(indexed_value).at(*span))
-                    }
-                    serde_json::Value::String(val) if val.is_empty() && is_safe => {
-                        Ok(DynVal::from(&serde_json::Value::Null).at(*span))
+                // Needs to be done first as `as_json_value` fails on empty string
+                if is_safe && val.as_string()?.is_empty() {
+                    Ok(DynVal::from(&serde_json::Value::Null).at(*span))
+                } else {
+                    match val.as_json_value()? {
+                        serde_json::Value::Array(val) => {
+                            let index = index.as_i32()?;
+                            let indexed_value = val.get(index as usize).unwrap_or(&serde_json::Value::Null);
```

Comment:
This isn't for this PR, but should we be throwing an error here? It doesn't seem like we'd want to silently ignore out of bounds accesses